docs/features/search/how-to-guides.md

```diff
@@ -276,9 +282,38 @@ const SearchPage = () => (
 export const searchPage = <SearchPage />;
 ```
 
-> **Important**: A default result item extension should be placed as the last child, so it can be used only when no other extensions match the result being rendered. If a non-default extension is specified, the `DefaultResultListItem` component will be used.
+> **Important**: A default result item extension (one that does not have a predicate) should be placed as the last child, so it can be used only when no other extensions match the result being rendered.
+> If a non-default extension is specified, the `DefaultResultListItem` component will be used.
+
+### 2. Custom search result extension in the SidebarSearchModal
+
+You may be using the SidebarSearchModal component. In this case, you can customize the search items in this component as follows:
+
+```tsx title="packages/app/src/components/Root/Root.tsx"
+import { SidebarSearchModal } from '@backstage/plugin-search';
+...
+export const Root = ({ children }: PropsWithChildren<{}>) => {
+  const styles = useStyles();
+
+  return <SidebarPage>
+    <Sidebar>
+      ...
+      <SidebarSearchModal searchResultChildren={[
```

Comment:
Prefixing the property name with search seems redundant. I suggest renaming it to `resultItems`.

Reply:
So my reasoning for calling it `searchResultChildren` is because it is the `children` of the `SearchResult` component. I wanted that to be explicit, so that the next devs know exactly where the children will go. I can certainly switch it but personally like really explicit property names.

Reply:
I see your point, let me share the reasoning behind my name suggestion, but feel free to decide 🙂:
I think we don't need to necessarily couple the property name with the implementation of the search modal, I mean, to me people using this component don't need to know how the property will be used internally by the search modal code. If the ‘SearchResult’ gets replaced in the search modal implementation in the future or stop receiving list items via children the name ´searchResultChildren’ will no longer make sense, but result items continue being a meanful name because it will continue be receiving a list of items independently if it is forwarded as a children or not.

Reply:
Okay I'll switch it, I get what you are saying now. I think both have benefits but also both make sense. The important thing is people can choose.

Reply:
I have a compromise suggestion. Rather than `resultItems` (since I would read this as the actual results themselves), how about `resultItemComponents`?

Reply:
That's a good idea, I like ’resultItemComponents’ 🙌